scala/src/main/scala/smile/clustering/distances/Binary_Distance.scala

```diff
@@ -1,22 +1,20 @@
 package smile.clustering.distances
 
-abstract class Binary_Distance extends Distance
-{
+abstract class BinaryDistance extends Distance {
```

Comment:
Is this class necessary? I understand there are some utility functions in the class. But it seems better to use this as mixin than a base class.